src/nyc_trees/apps/survey/templates/survey/partials/reserved_blockface_popup.html

```diff
@@ -13,13 +13,39 @@
             </button>
             <ul class="dropdown-menu dropdown-menu-right" role="menu" aria-labelledby="reserved-blockface-popup-submenu">
                 <li>
-                    <a role="menuitem" tabindex="-1"
-                      href="{% url 'survey' blockface_id=blockface.id %}">
+                    <a role="menuitem" tabindex="-1" href="{% url 'survey' blockface_id=blockface.id %}">
                         Start Mapping
                     </a>
                 </li>
-                <li><a role="menuitem" tabindex="-1" href="#">Cancel Reservation</a></li>
+                <li>
+                    <a role="menuitem" tabindex="-1" href="javascript:" id="cancel-reservation" data-toggle="modal" data-target="#reservation-modal">
+                      Cancel Reservation
+                    </a>
+                </li>
             </ul>
         </div>
     </div>
+    <!-- Bootstrap modals need to be high up in the DOM to prevent z-index issues,
+         but we want to send down new markup for each blockface.
+         The solution I arrived at is to place the modal contents in a script tag,
+         and move them into the modal when this partial is loaded -->
+    <script type="html/template" id="cancel-reservation-template">
+    <div class="modal-dialog">
+        <div class="modal-content">
+            <div class="modal-header">
+                <h4 class="modal-title" id="reservation-modal-label">Cancel Reservation</h4>
+            </div>
+            <div class="modal-body">
+                This reservation will expire on {{ reservation.expires_at|date }}.
```

Comment:
From a mappers perspective, being shown the expiration date in the confirmation dialog is a little confusing. What is the situation in which knowing the expiration date would change whether or not I follow through with my cancellation?